lib/std/testing.zig

```diff
@@ -435,3 +435,23 @@ pub fn refAllDecls(comptime T: type) void {
         _ = decl;
     }
 }
+
+/// Given a type, recursively reference all the declarations inside,
+/// so that the semantic analyzer sees them. Up to `levels` deep. 
+pub fn refAllDeclsInTree(comptime T: type, comptime levels: u8) void {
```

Comment:
I don't want to do this. This is a kludge working around the language shortcomings. I don't want people fighting with the compiler about recursive levels and trying to debug why stuff does or doesn't show up in the docs.

This function is fighting against the language but we need to take a step back and figure out how to make the language better with respect to auto generated docs. I'm not satisfied with this as a solution. I'd rather keep things painful and annoying until we have a proper solution.

Reply:
@andrewrk Is there an open issue for that? Im thinking of some new visibility keyword, for example `api` to be used instead of `pub` so its compiled even when not referenced (at least when generating docs). 

By the way, should existing `refAllDecs` be removed since its a hack?